dashboard/src/components/AppView/AppControls/StatusAwareButton.tsx

```diff
@@ -0,0 +1,32 @@
+import { CdsButton } from "@cds/react/button";
+import ReactTooltip from "react-tooltip";
+import { hapi } from "../../../shared/hapi/release";
+
+export interface IStatusAwareButtonProps {
+  id: string;
+  releaseStatus: hapi.release.IStatus | undefined | null;
+}
+
+export default function StatusAwareButton<T extends IStatusAwareButtonProps>(props: T) {
+  const { id, releaseStatus, ...otherProps } = props;
+  const disabled: boolean = releaseStatus?.code
```

Comment:
Currently, it isn't handling when it is loading and no status code has been received:

https://user-images.githubusercontent.com/11535726/123978094-de826300-d9bf-11eb-8c48-2b3e02a7e56a.mp4


https://user-images.githubusercontent.com/11535726/123978096-df1af980-d9bf-11eb-8df9-bbb05916a7c3.mp4

Reply:
I'd suggest sth like:

```js
  // Disable the button if: the code is undefined or null OR the status code is one of [5,6,7,8]
  // See https://github.com/kubeapps/kubeapps/blob/master/dashboard/src/shared/hapi/release.d.ts#L559
  const disabled = releaseStatus?.code ? _.inRange(releaseStatus?.code, 5, 9) : true;
```
See what you think

Reply:
in both videos, there is a status of code 1 (deployed) that is returned, that is why the buttons are not disabled. This is the issue i mentioned int he standup, the backend is not returning the proper code.
I was able to confirm that by always showing a tooltip with the code returned, and it is always "deployed".

unless we fix the backend, the UI will not be able to work as expected. I believe we agreed to wait until we move over to the new apis.

Reply:
regarding the second comment, getting a status code that is undefined is an edge case.
I think it is safer to not disable to avoid preventing the user from doing a compensating/fixing action.
but if you feel it is better to disable when the status is undefined, i can make the change.

Reply:
Exactly, the code: 1 is the one being returned usually. So there is no way for the frontend to be aware of that changes.
We can think of alternatives like adding a redux state action triggered when performing an upgrade, that would disable the button until a new status:1 event is returned again. This way we prevent users from clicking when the upgrade operation hasn't begun yet.
It's a bit more complicated and, given we are gonna refactor the backed, I don't think it worths it.

Regarding the `undefined` edge case: I don't see it as an edge case in that this status is returned every time the component is mounted (write a `console.log` statement and see how many times it gets called).
I think it would prevent users from clicking multiple times the buttons going back and forth on the browser's history (as I use to do when testing out an app haha).
(btw, doing so makes an error like this appear: https://stackoverflow.com/questions/53949393/cant-perform-a-react-state-update-on-an-unmounted-component)

In which situations do you think having the button enabled could be useful when the status is `undefined` ?

Reply:
i will disable the buttons when the status is undefined